### DISCUSSION TITLE
connection status problem

### DISCUSSION THREAD

**Discussion by DC-Lin** [miguelgrinberg/Flask-SocketIO#1838] [Q&A]
Please help, how can I know that the user closed the connection and then I stop sending content to the user

**Comment by DC-Lin** [comment]
For example, I keep sending messages to the user, but the user closed the connection, I have to stop sending messages to the user, what should I do to know the user closed the connection status

**Reply by miguelgrinberg** [reply]
Please do not write issues when you have questions. Use the discussion board instead.

When the user disconnects you receive the `disconnect` event.

**Reply by DC-Lin** [reply]
Thank you very much！

**Reply by DC-Lin** [reply]
Now, defining the disconnect event does not solve a problem. If disconnect is an event route, how does another route get the result returned by disconnect。
example:
@socketio.on('my_event')
def test_connect(auth):
    whilt not disconnect:    #if user don't closed this connection,but how can i get dicconnection flag here.
       emit('my response', {'data': 'Connected'})

@socketio.on('disconnect')
def test_disconnect():
    print('Client disconnected')

**Reply by miguelgrinberg** [reply]
This package does not support a serial way of writing your logic, you have to structure your application with events. You shouldn't have a while loop in the connect handler, that is an event in which you have to allow or reject clients, it should return quickly. I suggest you look at the examples in this repository to learn about creating events.

For example, you can set a global variable or event object in your disconnect handler, so that another function that is serving the client know when to stop.
